Makefile.core.mk

```
@@ -96,13 +96,13 @@ clean:
 	@rm -fr resources .htmlproofer tmp generated public out samples install go tests/integration/
 
 lint: clean_public build_nominify lint-copyright-banner lint-python lint-yaml lint-dockerfiles lint-scripts lint-sass lint-typescript lint-go
-	@scripts/lint_site.sh
+	@BRANCH=$(SOURCE_BRANCH_NAME) scripts/lint_site.sh
 
 lint-en: clean_public build_nominify lint-copyright-banner lint-python lint-yaml lint-dockerfiles lint-scripts lint-sass lint-typescript lint-go
 	@scripts/lint_site.sh en
 
 lint-fast:
-	@SKIP_LINK_CHECK=true scripts/lint_site.sh en
+	@BRANCH=$(SOURCE_BRANCH_NAME) SKIP_LINK_CHECK=true scripts/lint_site.sh
```

This is only used for local checks, BRANCH won't be set. So best to leave it as an en-only function.
```suggestion 
	SKIP_LINK_CHECK=true scripts/lint_site.sh en
```

---

But if they are using the makefile, BRANCH is set to the SOURCE_BRANCH_NAME so thinks should be fine. Especially for non-english devs. I would entertain an lint-fast-en if we wanted to force English.

---

If users want to run `lint-fast` on local changes, I don't think SOURCE_BRANCH_NAME default of `master` will do the right thing. The correct branch to use is `my-branch`, not `master`. That's why I think we should just do `en` for this, not require any extra setting of SOURCE_BRANCH_NAME. We also don't need to make it more complicated with `lint-fast-<lang>` rules because `en` is the only one that actually does any real spell checking, and the snip.sh lint checks are also only needed in the en tree.

---

@frankbu Ahh. Your scenario isn't correct. Let's say I have a branch with commits I have pushed to a PR or not. (ex: branch is lintByLanguage).

If I run the command against that branch with everything in a commit, I see:
```
git diff --name-only --diff-filter=b "$(git merge-base HEAD "lintByLanguage")"
```
If I instead run it against master, I get all the changes I have in the branch since it was created:
```
git diff --name-only --diff-filter=b "$(git merge-base HEAD "master")" 
Makefile.core.mk
scripts/lint_site.sh
```
I can get the current branch and try to use that in the script, but that never showed all my changes so didn't work. I've always run locally using `BRANCH=master ./scripts/lint_site.sh` to get the list of files. This is because my changes are typically in their own branch off of master created using something like `git checkout master` and then `git checkout -b lintByLanguage`

That said, if the person is doing some branching between PRs then they could see the list being generated incorrectly.

---

The above is why I defaulted to `master` as that would be the typical branch that someone would branch off of for their PRs.

---

I see, but defaulting to `master` won't do the right thing if my branch is off of branch `release-1.5`, for example. It won't fail, it just may not be doing the right thing. I still think that keeping `lint-fast` as a simple en-only function, for the reasons I mentioned above, is best. In fact, maybe `make lint` shouldn't default to master either, since that also may or may not be working properly in local.

---

But we removed any defaulting at this point. If one runs the script locally directly, they will get both languages as they would before this change. They can set BRANCH if they want if they want a faster scan.

If they are running the make file locally, they will use whatever branch is specified . If they are typically doing that in master, that's fine. If they breached of of release-1.5, then it will use releases-1.5 (it's all defined in the Makefile.core.mk that they are using as SOURCE_BRANCH_NAME ?= ...).